apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -1479,62 +1512,61 @@ channel_status() ->
         %% - connected:    the channel is added to the resource, the resource is
         %%                 connected and the on_channel_get_status callback has returned
         %%                 connected. The error field should be undefined.
-        status => ?status_disconnected,
-        error => not_added_yet
-    }.
-
-%% If the channel is added with add_channel/2, the config field will be set to
-%% the config. This is useful when doing probing since the config is not stored
-%% anywhere else in that case.
-channel_status_new_with_config(Config) ->
-    #{
         status => ?status_disconnected,
         error => not_added_yet,
-        config => Config
+        config => ChannelConfig
     }.
 
-channel_status_new_waiting_for_health_check() ->
+channel_status_new_waiting_for_health_check(ChannelConfig) ->
     #{
         status => ?status_connecting,
-        error => no_health_check_yet
+        error => no_health_check_yet,
+        config => ChannelConfig
     }.
 
-channel_status({?status_connecting, Error}) ->
+channel_status({?status_connecting, Error}, ChannelConfig) ->
     #{
         status => ?status_connecting,
-        error => Error
+        error => Error,
+        config => ChannelConfig
     };
-channel_status({?status_disconnected, Error}) ->
+channel_status({?status_disconnected, Error}, ChannelConfig) ->
     #{
         status => ?status_disconnected,
-        error => Error
+        error => Error,
+        config => ChannelConfig
     };
-channel_status(?status_disconnected) ->
+channel_status(?status_disconnected, ChannelConfig) ->
     #{
         status => ?status_disconnected,
-        error => <<"Disconnected for unknown reason">>
+        error => <<"Disconnected for unknown reason">>,
+        config => ChannelConfig
     };
-channel_status(?status_connecting) ->
+channel_status(?status_connecting, ChannelConfig) ->
     #{
         status => ?status_connecting,
-        error => <<"Not connected for unknown reason">>
+        error => <<"Not connected for unknown reason">>,
+        config => ChannelConfig
     };
-channel_status(?status_connected) ->
+channel_status(?status_connected, ChannelConfig) ->
     #{
         status => ?status_connected,
-        error => undefined
+        error => undefined,
+        config => ChannelConfig
     };
 %% Probably not so useful but it is permitted to set an error even when the
 %% status is connected
-channel_status({?status_connected, Error}) ->
+channel_status({?status_connected, Error}, ChannelConfig) ->
     #{
         status => ?status_connected,
-        error => Error
+        error => Error,
+        config => ChannelConfig
     };
-channel_status({error, Reason}) ->
+channel_status({error, Reason}, ChannelConfig) ->
     #{
         status => ?status_disconnected,
-        error => Reason
+        error => Reason,
+        config => ChannelConfig
```

Comment:
I believe we return this channel status in the APIs.  If so, we might want to redact this config in the API handlers, as it might contain sensitive data.  It might be already handled there, but worth confirming it.

Reply:
Good point. I will check.

Reply:
Configs are now removed from get_channel_status calls and the lookup calls: [0f10b0b](https://github.com/emqx/emqx/pull/13077/commits/0f10b0b6c33bdf3e8113d2952f23b5550ad9a60e)